Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/EntryDelegate.cpp

```diff
@@ -290,7 +290,9 @@ namespace AzToolsFramework
                 {
                     displayString = RichTextHighlighter::HighlightText(displayString, m_assetBrowserFilerModel->GetStringFilter()->GetFilterString());
                 }
+                painter->save();
                 RichTextHighlighter::PaintHighlightedRichText(displayString, painter, optionV4, remainingRect);
+                painter->restore();
```

Comment:
As mentioned in private, I would advise to keep the painter save/restore calls inside PaintHighlightedRichText, since the functions signature does not imply that the painter will be changed by the call.